test/acceptance/tests/basic/basic_test.go

```diff
@@ -352,217 +352,306 @@ func TestValidation_ConsulEcsConfigVariable(t *testing.T) {
 	}
 }
 
+// Test the validation that if a partition is provided, a namespace must also be provided.
+func TestValidation_NamespaceIsRequiredIfPartitionIsSet(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		partition string
+		namespace string
+		errMsg    string
+	}{
+		"without partition and namespace": {
+			partition: "",
+			namespace: "",
+			errMsg:    "",
+		},
+		"with partition and namespace": {
+			partition: "default",
+			namespace: "default",
+			errMsg:    "",
+		},
+		"with partition, without namespace": {
+			partition: "default",
+			namespace: "",
+			errMsg:    "ERROR: consul_namespace must be set if consul_partition is set",
+		},
+		"without partition, with namespace": {
+			partition: "",
+			namespace: "default",
+			errMsg:    "",
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+				TerraformDir: "./terraform/admin-partition-validate",
+				NoColor:      true,
+				Vars: map[string]interface{}{
+					"partition": c.partition,
+					"namespace": c.namespace,
+				},
+			})
+			t.Cleanup(func() {
+				_, _ = terraform.DestroyE(t, terraformOptions)
+			})
+			_, err := terraform.InitAndPlanE(t, terraformOptions)
+			if c.errMsg == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), c.errMsg)
+			}
+		})
+	}
+}
+
 func TestBasic(t *testing.T) {
-	randomSuffix := strings.ToLower(random.UniqueId())
-	tfVars := suite.Config().TFVars("route_table_ids")
-	tfVars["suffix"] = randomSuffix
-	serverServiceName := "custom_test_server"
-	// This uses the explicitly passed service name rather than the task's family name.
-	tfVars["server_service_name"] = serverServiceName
-	clientServiceName := "test_client"
+	cases := []bool{false, true}
+
+	for _, secure := range cases {
+		t.Run(fmt.Sprintf("secure: %t", secure), func(t *testing.T) {
+			randomSuffix := strings.ToLower(random.UniqueId())
+			tfVars := suite.Config().TFVars("route_table_ids")
+			tfVars["secure"] = secure
+			tfVars["suffix"] = randomSuffix
+			clientServiceName := "test_client"
+
+			serverServiceName := "test_server"
+			if !secure {
+				// This uses the explicitly passed service name rather than the task's family name.
+				serverServiceName = "custom_test_server"
```

Comment:
Why is this service name different for insecure vs secure?

Reply:
The "insecure" case had it set to an explicit value and I thought it would be more complete to test with different permutations for service name. I can set them both to `custom_test_server` if that makes more sense.

Reply:
👍 That makes sense.

(Just wanted to check there was not some odd reason this was required to get the secure case passing.)

Reply:
I swapped the `serverServiceName` for the secure and insecure case and the test failed. The `secret_name` is using `var.family` but it should be using `local.service_name`. So the secure case was only passing coincidentally because `test_server` is the same as the `family`.